pkg/inject/sidecar.go

```diff
@@ -378,3 +386,12 @@ func EqualSidecar(dep, oldDep *appsv1.Deployment) bool {
 	oldDepContainer := oldDep.Spec.Template.Spec.Containers[oldDepIndex]
 	return reflect.DeepEqual(depContainer, oldDepContainer)
 }
+
+func parseAgentTags(args []string) []string {
```

Comment:
If we follow the refactoring mentioned above, we might not need this. The reason is that we'd have a consistent result for the map, as long as the container name doesn't change. Once it changes, we want this new name to be used in the map/args.

Reply:
I partially agree for this point. Suppose we have a deployment with one single container at the very beginning, of course we can readily deduce the `container.name`. Then we inject several sidecars base on the annotations, let's say, istio-proxy and jaeger-agent, we will finally have three containers in this deployment.

At the second time, even if we are able to filter the jaeger-agent out, we still have two containers which we are not able to pick the correct `container.name` without a prior. So I think it is still necessary to parse the Tags / Args from the existing agent container.

Reply:
We should really leave the `container.name` out if we have more than one container (not counting the Jaeger Agent one). It's certainly possible that traces are coming from both containers in that case, so, the annotation will be wrong. Reusing the container name from an older version of the pod doesn't seem right to me.

Reply:
> We should really leave the `container.name` out if we have more than one container (not counting the Jaeger Agent one). It's certainly possible that traces are coming from both containers in that case, so, the annotation will be wrong. Reusing the container name from an older version of the pod doesn't seem right to me.

I understand your argument and I think it is quite reasonable. But the deployment may still probably restart once if we inject more than one sidecar. That would cause a serious problem in a large cluster (we have ~500 deployments in our production cluster). How can we prevent from this situation?

Reply:
This will happen only if you didn't have another container before, but now you have, right? The only way to prevent that, from what I can see, is to add the agent sidecar manually, so that the operator won't manage this container for you.

Reply:
Specifically, I am concerned with the scenario with two operators: one is `jaeger-operator`, the other is `istio-operator`. 

Emm, what about adding an annotation to specify the container-name, if the user want to set the name explicitly.

Reply:
Users can do that already by adding env vars to their own container (JAEGER_TAGS). The tags we add to the sidecar container are ones that we add based on information we can infer from the env.